elasticgraph-graphql/lib/elastic_graph/graphql/filtering/filter_node_interpreter.rb

```diff
@@ -25,22 +25,32 @@ def initialize(runtime_metadata:)
         end
 
         def identify_node_type(field_or_op, sub_expression)
-          return :empty if sub_expression.nil? || sub_expression == {}
+          identify_by_field_or_op(field_or_op) || identify_by_sub_expr(sub_expression) || :unknown
+        end
+
+        def filter_operators
+          @filter_operators ||= build_filter_operators(runtime_metadata)
+        end
+
+        private
+
+        def identify_by_field_or_op(field_or_op)
           return :not if field_or_op == schema_names.not
           return :list_any_filter if field_or_op == schema_names.any_satisfy
           return :all_of if field_or_op == schema_names.all_of
           return :any_of if field_or_op == schema_names.any_of
           return :operator if filter_operators.key?(field_or_op)
           return :list_count if field_or_op == LIST_COUNTS_FIELD
-          return :sub_field if sub_expression.is_a?(::Hash)
-          :unknown
-        end
 
-        def filter_operators
-          @filter_operators ||= build_filter_operators(runtime_metadata)
+          nil
         end
 
-        private
+        def identify_by_sub_expr(sub_expression)
+          return :empty if sub_expression.nil? || sub_expression == {}
+          return :sub_field if sub_expression.is_a?(::Hash)
+
+          nil
```

Comment:
```suggestion
```

`nil` is the automatic return value when no branches are taken.

If you like this suggestion, please apply it to `identify_by_field_or_op` as well.

Reply:
Nevermind, we discussed how rubocop requires this.